spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteReachableFilesSparkAction.java

```diff
@@ -132,7 +126,20 @@ private Dataset<FileInfo> reachableFileDS(TableMetadata metadata) {
   }
 
   private DeleteReachableFiles.Result deleteFiles(Iterator<FileInfo> files) {
-    DeleteSummary summary = deleteFiles(deleteExecutorService, deleteFunc, files);
+    DeleteSummary summary;
+    if (deleteFunc == null && io instanceof SupportsBulkOperations) {
+      summary = deleteFiles((SupportsBulkOperations) io, files);
+    } else {
+
```

Comment:
I actually meant an empty line after `DeleteSummary` var but formatting here is up to you.
I like the new pattern.